mapper/tedge_mapper/src/sm_c8y_mapper/mapper.rs

```diff
@@ -105,12 +107,23 @@ impl CumulocitySoftwareManagement {
     ) -> Result<(), SMCumulocityMapperError> {
         let response = SoftwareListResponse::from_json(json_response)?;
 
+        let topic = OutgoingTopic::SmartRestResponse.to_topic()?;
+        let smartrest_response =
+            SmartRestSetSoftwareList::from_thin_edge_json(response).to_smartrest()?;
+        let () = self.publish(&topic, smartrest_response).await?;
+
+        Ok(())
+    }
+
+    async fn validate_and_publish_software_list(
+        &self,
+        json_response: &str,
+    ) -> Result<(), SMCumulocityMapperError> {
+        let response = SoftwareListResponse::from_json(json_response)?;
+
         match response.status() {
             SoftwareOperationStatus::Successful => {
-                let topic = OutgoingTopic::SmartRestResponse.to_topic()?;
-                let smartrest_response =
-                    SmartRestSetSoftwareList::from_thin_edge_json(response).to_smartrest()?;
-                let () = self.publish(&topic, smartrest_response).await?;
+                self.publish_software_list(json_response).await?;
```

Comment:
Is it intended to only send the list when successful and do nothing on Failed?

Reply:
Yes, this function is called only for `tedge/commands/res/software/list` topic. It's not an operation for c8y. Mapper should discard failed or executing for software list, unlike software update.